dice-game-ui-angular/src/app/game/game.component.ts

```diff
@@ -16,54 +17,58 @@ interface FieldPoint {
   selector: 'app-game',
   templateUrl: './game.component.html',
   styleUrls: ['./game.component.scss'],
-  providers: [{ provide: GameStorageService, useClass: LocalGameStorageService }],
 })
 export class GameComponent implements OnInit {
   private readonly setColor: string = '#D28EFF';
   private readonly selectedColor: string = '#3f51b5';
   private readonly emptyColor: string = 'lightblue';
   private canvasMaxHeight = 600;
 
+  private gameGuid: Guid;
   private gameEngine: GameEngine;
   private simpleDice: Dice;
 
   private points: FieldPoint[] = [];
 
   private selecting = false;
-  private svg = null;
+  private svg: any = null;
   private startPoint: FieldPoint = null;
 
-  public dice1 = 0;
-  public dice2 = 0;
-  public gameFinished = false;
-  public score = 0;
+  dice1 = 0;
+  dice2 = 0;
+  gameFinished = false;
+  score = 0;
 
-  constructor(private gameStorageService: GameStorageService) {
+  constructor(private router: Router, private route: ActivatedRoute, private gameStorageService: GameStorageService) {
     this.simpleDice = new SimpleDice(6);
+  }
+
+  ngOnInit(): void {
+    this.gameGuid = Guid.parse(this.route.snapshot.paramMap.get('id'));
     try {
-      this.gameEngine = gameStorageService.restoreGame();
+      this.gameEngine = this.gameStorageService.restoreGame(this.gameGuid);
     } catch {
       throw new Error('The game was not started.');
     }
 
     if (this.gameEngine) {
       this.gameEngine.registerOnStateChanged((engine) => {
-        gameStorageService.saveGame(this.gameEngine);
+        this.gameStorageService.saveGame(this.gameEngine, this.gameGuid);
```

Comment:
According to implementation of storage service you save only field size of game engine. So you don't need to save game when state is changed, because size is not changed ever.

Reply:
It's a temporary thing due to luck of this functionality inside game engine. I'll fix it when we do these changes

Reply:
The service will save the enitre game engine object but it is not a problem of the game component

Reply:
You're right, agree with you.